backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/controller/ConfirmationCodesController.java

```diff
@@ -41,18 +42,21 @@ public ConfirmationCodesController(UserService userService) {
 	}
 
 	@GetMapping
+	@ApiResponse(responseCode = "405", description = "Method Not Allowed")
 	@Operation(summary = "List all confirmation codes for a user")
 	public CollectionModel<ConfirmationCodeModel> getConfirmationCodesByUserId(
 			@NotBlank(message = "userId must not be null or blank")
 			@Parameter(description = "The ID of the user.", example = "00000000-0000-0000-0000-000000000000", required = true)
 			@PathVariable String userId) {
-		userService.getUserById(userId)
+		/*userService.getUserById(userId)
 			.orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(userId)));
 
 		return userService.getUserById(userId)
 			.map(User::getConfirmationCodes)
 			.map(confirmationCodes -> confirmationCodeModelMapper.toModel(userId, confirmationCodes))
-			.orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(userId)));			
+			.orElseThrow(() -> new ResourceNotFoundException("No user with id=[%s] was found".formatted(userId))); */
+
+		throw new MethodNotAllowedException("API calls to this endpoint are not allowed.");
```

Comment:
I'm wondering if simply returning `void` with a `@ResponseStatus` would be sufficient?

Also I think we should log a warning when the method is fired.

Reply:
I tested this in the API and it works.

``` java
@GetMapping("/foo")
@ResponseStatus(HttpStatus.METHOD_NOT_ALLOWED)
void foo() {}
```

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/d53f263f-07b5-41fa-95dd-83c94fb44c90)

Reply:
I wondered the same thing so I too tried.

The `linkTo(..)` below expects a non-void argument:

https://github.com/DTS-STN/canadian-dental-care-plan/blob/3aa7a7cbb590b77e8d52851b263fc8a3f4063a70/backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/UserModelMapper.java#L48

Reply:
Right, I was referring to the return type as well. 

To add a `link` to the `confirmation-codes` resource, the mapper (specifically the snippet I pasted above) needs a `linkTo(..)` that references a method that is not `void`.

Reply:
added log.warn(...)

Reply:
``` java
@GetMapping
@ResponseStatus(HttpStatus.METHOD_NOT_ALLOWED)
@Operation(summary = "List all confirmation codes for a user")
public ResponseEntity<Void> getConfirmationCodesByUserId(
		@NotBlank(message = "userId must not be null or blank")
		@Parameter(description = "The ID of the user.", example = "00000000-0000-0000-0000-000000000000", required = true)
		@PathVariable String userId) {
	return ResponseEntity.status(HttpStatus.METHOD_NOT_ALLOWED).build();
}
```

Reply:
That works because we return `ResponseEntity<Void>`. If we go with this solution, we might as well delete this `@GetMapping` method altogether and have the `@DeleteMapping` return `ResponseEntity<Void>`. Then we can have the mapper add the `linkTo(..)` the `@DeleteMapping` method.

Reply:
^ Sorry, by `@DeleteMapping`, I meant `@PostMapping`, which is a `void` method. This controller has no `@DeleteMapping`.

Reply:
So the suggestion is to `linkTo()` the `createConfirmationCodeForUser(..)` method, which would have to be modified to return a `ResponseEntity<Void>` so the hateoas link build doesn't barf.

Correct, @nicholas-ly?

Reply:
Correct

Reply:
fixed